x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/build_exceptions_query.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { isEmpty } from 'lodash';
+
+import { Query } from '../../../../../../../../src/plugins/data/server';
+import { List, ListOperator, ListValues } from '../routes/schemas/types/lists_default_array';
+import { RuleAlertParams } from '../types';
+
+export const getLanguageBooleanOperator = (language: string, value: string) => {
```

Comment:
Can't we just restrict this to being the smallest set of lucene from here?

```ts
https://lucene.apache.org/solr/guide/6_6/the-standard-query-parser.html#TheStandardQueryParser-BooleanOperatorsSupportedbytheStandardQueryParser
```

Which would be `AND`, `NOT`, `OR`? 

```ts
value: `not | or | and`
```

```ts
export const getLanguageBooleanOperator = (language: string, value: 'and' | 'or' | 'not') => {
```

Same goes for using `language` everywhere. Can we restrict that within the types to being `lucene | kql | ...` ?

Makes it easier to maintain and work with the code in the long run. Makes it more understandable about why we are capitalizing these characters and not all characters when lucene.

Reply:
Your tests will break:

```ts
const result = getLanguageBooleanOperator('lucene', 'jibberjabber');
```

But those don't show me a clear example as it is telling me that it is getting the language boolean operator but its  example is not using the boolean operators which kind of made me scratch my head a bit.

Reply:
Now that you say it, I'm scratching my head a bit too :) I updated the types to reflect expectations. Changed `language: string | null | undefined` to be `language: keyof Language | null | undefined` where `Language` is `{ lucene: string; kuery: string }`. Updated tests as well.